reactor-netty-core/src/main/java/reactor/netty/transport/ClientTransport.java

```diff
@@ -250,6 +251,31 @@ public T proxy(Consumer<? super ProxyProvider.TypeSpec> proxyOptions) {
 		return dup;
 	}
 
+	/**
+	 * Set up proxy from java system properties.
+	 * Supports http, https, socks4, socks5 proxies.
+	 * List of supported system properties https://docs.oracle.com/javase/7/docs/api/java/net/doc-files/net-properties.html
+	 *
+	 * @return a new {@link ClientTransport} reference
+	 */
+	public final T proxyWithSystemProperties() {
+		return proxyWithSystemProperties(System.getProperties());
+	}
+
+	/**
+	 * Same as proxyWithSystemProperties() but accepts properties and used in testing only.
+	 *
+	 * @return a new {@link ClientTransport} reference
+	 */
+	final T proxyWithSystemProperties(Properties properties) {
+		T dup = duplicate();
+		ProxyProvider proxy = ProxyProvider.createFrom(properties);
+		if (proxy != null) {
+			dup.configuration().proxyProvider(proxy);
+		}
```

Comment:
good questions. I guess, there are two strategies: delete the previous config completely and replace it with the new one from system properties or to merge both configs and use values from system properties to override already defined values.
I'd go with the first approach because it is simpler and more clear, plus it does the same as 
```java
TcpClient client1 = TcpClient.create().proxy(....);
TcpClient client2 = client1.proxy(...);
```

I added this use case in the method's javadoc.
Are you OK with the first overriding ProxyProvider?

Reply:
> Are you OK with the first overriding ProxyProvider?

Yes agree.
Even when `ProxyProvider.createFrom(properties)` produces NULL we should override. Correct?

Reply:
Yes agree.
Even when ProxyProvider.createFrom(properties) produces NULL we should override. Correct?

Reply:
I think that we still need to address this use case when there are no proxy settings.

Let's have this test:

```
	@Test
	void test() {
		TestClientTransport transport = createTestTransportForProxy();
		transport.proxy(spec -> spec.type(ProxyProvider.Proxy.HTTP).host("proxy").port(8080));
		assertThat(transport.configuration().proxyProvider).isNotNull();

		transport.proxyWithSystemProperties(new Properties());
		assertThat(transport.configuration().proxyProvider).isNull();
	}
```

It should pass, but currently we still check for null and do not apply the changes, the proxy stays as it was defined previously:

```
	if (proxy != null) {
		dup.configuration().proxyProvider(proxy);
	}
```

Reply:
good point, missed that one. Removed the null check and added the test. Thanks.